streamer/src/nonblocking/quic.rs

```diff
@@ -187,6 +187,7 @@ async fn handle_connection(
                             break;
                         }
                     }
+                    stats.total_streams.fetch_sub(1, Ordering::Relaxed);
```

Comment:
This looks good but it's not sufficient to fix the `total_streams` count. Can you take a look at the other places where `stats.total_streams.fetch_sub(1, Ordering::Relaxed);` is called to ensure that we're never decrementing when we shouldn't?